tiled gallery: carousel-image-args.php: Fix PHP warnings from nested arrays in image meta

mreishus opened the pull request:
`carousel-image-args.php` can create PHP warnings when encoding image metadata with containing nested arrays.  The line responsible is:
```php
$fuzzy_image_meta = wp_json_encode( array_map( 'strval', $fuzzy_image_meta ),  ... 
```

An example $fuzzy_image_meta triggering the warning is:
```
Array                                                                          
(                                      
    [aperture] => 74099370.67          
    [credit] =>                        
    [camera] => aterma                                                         
    [caption] =>                       
    [created_timestamp] =>             
    [copyright] =>                     
    [focal_length] => 74099370.666667                                          
    [iso] => Array                     
        (                              
            [0] => 64                  
            [1] => 64                  
        )                              
                                                                               
    [shutter_speed] => 74099370.666667 
    [title] =>                         
)  
```

The warning is: ` Warning: Array to string conversion in (...)/modules/tiled-gallery/tiled-gallery/templates/partials/carousel-image-args.php on line 15 `.

Because `$fuzzy_image_meta['iso']` is an array, it gets converted to `Array` when running strval on it.
Solution: Create a new recursive function for strval, that aligns with the original intention of changing numbers to strings, but also handles nested arrays.

## Proposed changes:

* Instead of using `wp_json_encode( array_map( 'strval', $fuzzy_image_meta )` to encode image meta, use `$fuzzy_image_meta = wp_json_encode( Jetpack_Tiled_Gallery_Layout::stringify_array_values( $fuzzy_image_meta )` after introducing a new function that handles nested arrays correctly.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->

## Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* See D136291-code for a WPCOM testing helper and testing instructions.

mreishus committed:
tiled gallery: carousel-image-args.php: Fix PHP warnings from nested arrays in image meta

mreishus committed:
changelog

anomiex reviewed (commented):
The whole situation here seems confusing.

It looks like the only reason we have the string cast in the first place is because, back in 2012 (3952-wpcom), WordPress stopped casting the floats to strings in the exif data and instead of fixing the JS that was choking on non-strings we just added the casts in the PHP. Maybe it's finally time to change the `! val.length` [here](https://github.com/Automattic/jetpack/blob/cda58d72b690fc02aae37039bf0f7810397df28e/projects/plugins/jetpack/modules/carousel/jetpack-carousel.js#L1067) to `! val` instead?

I also note that passing through the array would still do the wrong thing in the JS at [line 1083](https://github.com/Automattic/jetpack/blob/cda58d72b690fc02aae37039bf0f7810397df28e/projects/plugins/jetpack/modules/carousel/jetpack-carousel.js#L1083), except it should never get there because "iso" isn't one of [the whitelisted keys](https://github.com/Automattic/jetpack/blob/cda58d72b690fc02aae37039bf0f7810397df28e/projects/plugins/jetpack/modules/carousel/jetpack-carousel.php#L474). So at least we don't have to worry about that.

If we do decide to keep the cast, at least for now, it might be simpler to just do `map_deep( $fuzzy_image_meta, 'strval' )` ([doc](https://developer.wordpress.org/reference/functions/map_deep/)) instead of defining our own function.

mreishus committed:
Remove stringify_array_values and use map_deep instead

mreishus commented:
> The whole situation here seems confusing.
> 
> It looks like the only reason we have the string cast in the first place is because, back in 2012 (3952-wpcom), WordPress stopped casting the floats to strings in the exif data and instead of fixing the JS that was choking on non-strings we just added the casts in the PHP. Maybe it's finally time to change the `! val.length` [here](https://github.com/Automattic/jetpack/blob/cda58d72b690fc02aae37039bf0f7810397df28e/projects/plugins/jetpack/modules/carousel/jetpack-carousel.js#L1067) to `! val` instead?
> 
> I also note that passing through the array would still do the wrong thing in the JS at [line 1083](https://github.com/Automattic/jetpack/blob/cda58d72b690fc02aae37039bf0f7810397df28e/projects/plugins/jetpack/modules/carousel/jetpack-carousel.js#L1083), except it should never get there because "iso" isn't one of [the whitelisted keys](https://github.com/Automattic/jetpack/blob/cda58d72b690fc02aae37039bf0f7810397df28e/projects/plugins/jetpack/modules/carousel/jetpack-carousel.php#L474). So at least we don't have to worry about that.
> 
> If we do decide to keep the cast, at least for now, it might be simpler to just do `map_deep( $fuzzy_image_meta, 'strval' )` ([doc](https://developer.wordpress.org/reference/functions/map_deep/)) instead of defining our own function.

Thanks for the research and insight. I don't feel too comfortable changing the behavior as I don't know that much about the carousel and how it works. For solving PHP 8 warnings, I'm leaning towards making a low risk change that doesn't modify behavior. I've updated to use `map_deep()`, which is better than the one-off function I had before. :)

anomiex reviewed (approved):
Thanks!
